### PR TITLE
Add error handling for 500 responses

### PR DESCRIPTION
**Pull request by jtjobe**
Also, this PR for a related change to Visualize.

https://github.com/hyp3rventures/cc-editor-api/pull/111

### CODE REVIEW

File: spec/authentication_service/request_spec.rb [hunk 1]
```diff
@@ -65,7 +65,7 @@
         let(:status) { { status: 401, body: result.to_json } }
 
         it 'raises an UnauthorizedUserError' do
-          expect { subject.run(user) }.to raise_error(Hyper::AuthenticationService::UnauthorizedUserError)
+          expect { subject.run(user) }.to raise_error(Hyper::AuthenticationService::UnauthorizedUserError, result["error"])
```
Comment (dvandersluis) [inline spec/authentication_service/request_spec.rb]: Looks like we don't have rubocop set up for this repo, but strings should be single quoted unless double quotes are necessary

File: lib/hyper/authentication_service/request.rb [hunk 2]
```diff
@@ -27,6 +25,11 @@ def headers_for(user)
       rescue KeyError => e
         raise InvalidUserObjectError, e.message
       end
+
+      def handle_errors(response)
+        raise InternalServerError if response.status == 500
+        raise UnauthorizedUserError, JSON.parse(response.body)['error'] if response.body['error']
+      end
```
Comment (dvandersluis) [inline lib/hyper/authentication_service/request.rb]: ❤️

**Review by dvandersluis — APPROVED** [review]
One minor change please.

File: lib/hyper/authentication_service/request.rb [hunk 3]
```diff
@@ -27,6 +25,11 @@ def headers_for(user)
       rescue KeyError => e
         raise InvalidUserObjectError, e.message
       end
+
+      def handle_errors(response)
+        raise InternalServerError if response.status == 500
```
Comment (dvandersluis) [inline lib/hyper/authentication_service/request.rb]: @flanger001 are we concerned about other 5xx statuses?

File: spec/authentication_service/request_spec.rb (hunk 1 above)
Comment (flanger001) [inline spec/authentication_service/request_spec.rb]: Let's make this single quotes e.g. `result['error']`

File: lib/hyper/authentication_service/request.rb (hunk 3 above)
Comment (flanger001) [inline lib/hyper/authentication_service/request.rb]: Can we add more errors to this? 502 and 504 happen quite a bit as well.

File: spec/authentication_service/request_spec.rb [hunk 4]
```diff
@@ -77,5 +77,13 @@
         expect { subject.run(user) }.to raise_error(Hyper::AuthenticationService::InvalidUserObjectError)
       end
     end
+
+    context '500 response code' do
```
Comment (flanger001) [inline spec/authentication_service/request_spec.rb]: Suggestion: can we make this have a slightly more descriptive message?

**Review by flanger001 — CHANGES_REQUESTED** [review]
Small changes, looks good otherwise!

File: lib/hyper/authentication_service/request.rb (hunk 3 above)
Comment (flanger001) [reply lib/hyper/authentication_service/request.rb]: https://github.com/hyp3rventures/authentication_service/pull/7#discussion_r172287471

**Review by flanger001 — APPROVED** [review]

**Pull request merged**